### PR TITLE
Display all authors on Research page's authors filter tab

### LINKED ISSUE #1554: Not all authors are displayed properly on Research page's authors filter tab
This is visible in the video below. When trying to filter by author, the last author to appear in the list is Donglai Xu, but there are about five authors below that one, and when a user scrolls all the way to the very bottom of the page, they can see them, but otherwise, it's impossible to apply filters with these authors.

https://github.com/PolicyEngine/policyengine-app/assets/14987227/24f6e956-42fa-43ea-8455-b90d4ee8db78

### PR DESCRIPTION
**Pull request by NishantSatere**
## Description

Fixes #1554

## Changes

implemented a scrollable checkbox list and also added a feature where opening one checkbox list closes the others

##Screenshots

https://github.com/PolicyEngine/policyengine-app/assets/100207648/f359f978-e5ac-4c9e-b5f8-035f36525143

### CODE REVIEW

**Comment by anth-volk** [comment]
Just one thing before I review later today: Could you change the title to what this does? E.g., "Display all authors on Research page's authors filter tab." Your current format is what we use for the issues, not the fixes

**Comment by NishantSatere** [comment]
I am getting this error
Screenshot 2024-04-11 at 7 40 47 PM

**Comment by NishantSatere** [comment]
yes!

**Comment by anth-volk** [comment]
In that case, could you just run `npm run fix` and see if that solves it? Also, if you could check out the merge conflict in `src/redesign/components/Research.jsx`, that should be about it.

**Comment by NishantSatere** [comment]
i have fixed the lint issue and merge conflict

File: src/redesign/components/Research.jsx [hunk 1]
```diff
@@ -343,59 +356,63 @@ function ExpandableCheckBoxList({
...
-      />
+      </div>
     </Expandable>
   );
 }
 
-function Expandable({ title, children }) {
-  const [expanded, setExpanded] = useState(false);
+function Expandable({ title, children, isOpen, onToggle }) {
+  // const [expanded, setExpanded] = useState(false);
```
Comment (anth-volk) [inline src/redesign/components/Research.jsx:400]:
```suggestion
```

**Review by anth-volk — CHANGES_REQUESTED** [review]
Unfortunately, it looks like this doesn't quite solve the issue. On most desktop displays, this works fine, but imagine a display that's very short. Unfortunately, the bottom of the list is still cut off. The video below demonstrates this behavior.

As such, could you find a way to either: 
1. Make sure the list only expands to the bottom of the page, or 
2. Allow the entirety of all menus to open simultaneously, and just add a scroll capability there (this is probably the better one)

Sorry, this didn't come across in your earlier video, but it could be an issue for some users.


https://github.com/PolicyEngine/policyengine-app/assets/14987227/60cf74c1-46a9-4fc5-9a26-38ceef096a2d

**Comment by NishantSatere** [comment]
i have made changes !

**Comment by anth-volk** [comment]
Great! I will review first thing Monday. Thanks a lot, @NishantSatere.

**Review by anth-volk — CHANGES_REQUESTED** [review]
Thanks so much for this, @NishantSatere. While this is an improvement over the previous changes, this still has one particular issue: if the user has a short viewport, the bottom authors are still cut off. I demonstrate this in the video below.

https://github.com/PolicyEngine/policyengine-app/assets/14987227/09787bf6-07fc-4ff8-951d-40b9d51aab58

I would encourage you to consider different viewport sizes as you make edits, as there's no longer any guarantee that users will have a particular screen size. In the case of these edits, I'd explore using a percentage-based height and potentially editing the height of the parent container to ensure that the menu never extends beyond the height of the screen.

**Comment by anth-volk** [comment]
@NishantSatere If you're still working on this, and not totally deleting your work, I would encourage converting this to a "draft" PR. There's a small link in the top-right of the PR page to do so that says, "Still in progress? Convert to draft". When you close the PR fully, it signals to us that you're either changing it very heavily or you no longer want to work on the issue.

**Pull request closed without merging**